submodules/power_management_services_setup/main.tf

```diff
@@ -0,0 +1,108 @@
+#####################################################
+# Configure Squid client
+#####################################################
+
+resource "null_resource" "perform_proxy_client_setup" {
+
+  count = var.perform_proxy_client_setup != null ? length(var.perform_proxy_client_setup["squid_client_ips"]) : 0
+
+  connection {
+    type         = "ssh"
+    user         = "root"
+    bastion_host = var.access_host_or_ip
```

Comment:
It'd be desirable to add bastion_user and bastion_private_key in the block, so that users can configure different keys for bastion and remove host, and different user. This would come handy to further segregate access.

Reply:
in general, good to segregate, but:
- if we would use another user, we must create it from root first
- the whole SAP installation is executed as root

Reply:
Ok let's review in future when the case emerges